src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -332,9 +332,10 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
             }
 
             $getValue = function () use ($type, $parameter, $class, $method) {
-                if (!$value = $this->getAutowiredReference($ref = new TypedReference($type, $type, ContainerBuilder::EXCEPTION_ON_INVALID_REFERENCE, Target::parseName($parameter)), false)) {
+                $name = Target::parseName($parameter, $target);
+                if (!$value = $this->getAutowiredReference($ref = new TypedReference($type, $type, ContainerBuilder::EXCEPTION_ON_INVALID_REFERENCE, $name, $target ? [$target] : []), false)) {
                     $failureMessage = $this->createTypeNotFoundMessageCallback($ref, sprintf('argument "$%s" of method "%s()"', $parameter->name, $class !== $this->currentId ? $class.'::'.$method : $method));
-
+                   
```

Comment:
white space must be removed :)

Reply:
Thanks for your patience and all comments!!!
Not my best day, but the next PR should be better